runtime/gc_structs/ConstantPoolConstantDynamicSlotIterator.hpp

```diff
@@ -0,0 +1,83 @@
+
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup GC_Structs
+ */
+
+#if !defined(CONSTANTPOOLCONSTANTDYNAMICSLOTITERATOR_HPP_)
+#define CONSTANTPOOLCONSTANTDYNAMICSLOTITERATOR_HPP_
+
+#include "j9.h"
+#include "j9cfg.h"
+#include "j9cp.h"
+#include "modron.h"
+#include "j2sever.h"
+
+#include "EnvironmentBase.hpp"
+
+/**
+ * Iterate over ConstantDynamic references in the constant pool of a class.
+ * 
+ * @ingroup GC_Structs
+ */
+class GC_ConstantPoolConstantDynamicSlotIterator
+{
+
+	J9Object **_cpEntry;
+	U_32 _cpEntryCount;
+	U_32 _cpEntryTotal;
+	U_32 *_cpDescriptionSlots;
+	U_32 _cpDescription;
+	UDATA _cpDescriptionIndex;
+	
+public:
+	GC_ConstantPoolConstantDynamicSlotIterator(MM_EnvironmentBase *env, J9Class *clazz) :
+		_cpEntry((J9Object **)J9_CP_FROM_CLASS(clazz)),
+		_cpEntryCount(clazz->romClass->ramConstantPoolCount)
+	{
+		if(J2SE_VERSION((J9JavaVM*)env->getLanguageVM()) < J2SE_V11) {
```

Comment:
add a comment here saying:
```
TODO: detect whether class can have any Condy entries
```

Reply:
Also update the commit message saying this is the firs part of two changes, the next change will check if the class contains any Condy cp entries